### ISSUE TITLE
Some problems with route authorization

### ISSUE THREAD

**Issue by anthonychu** [Azure/static-web-apps-cli#174]
**To Reproduce**
https://github.com/anthonychu/anthonychu-test-vueapp2/blob/ad07cbd66c5845b28a5b4b2118c28f98f6bee6c4/staticwebapp.config.json

Clone above repo

```
npm i
npm run build
cd api
npm i
cd ..
swa start dist --api api
```

Go to `http://localhost:4280` and log in.
`/.auth/me` verifies logged in

* `/` still returning 401
* `/api/info` still returning 401

CLI v0.3.0

**Comment by manekinekko** [comment]
@anthonychu can you confirm that this is what you expect:
![image](https://user-images.githubusercontent.com/1699357/114197679-dd0e4500-9952-11eb-8211-793c5fb9a06c.png)

**Comment by anthonychu** [comment]
Look's great. Thanks. I added a suggestion in the PR to make the test descriptions more clear.

**Comment by anthonychu** [comment]
@manekinekko  This didn't seem to fix the issue for me. Can you try cloning the repo and see if you can reproduce with the latest build from main?

**Comment by manekinekko** [comment]
Sure. I am Investigating this rn.

**Comment by manekinekko** [comment]
@anthonychu in your config file, you are restricting the `/` to roles with `authenticated` only:

```
{
    "route": "/",
    "allowedRoles": ["authenticated"]
},
```

So, we are blocking access to `/` because the user is not logged in.

**Comment by anthonychu** [comment]
It still returns 401 after logging in though (cookie is set). Here’s the same app deployed:

this is 401:
https://blue-cliff-01805301e.azurestaticapps.net/

then log in:
https://blue-cliff-01805301e.azurestaticapps.net/.auth/login/github

and now the routes work.

**Comment by manekinekko** [comment]
Oh! It looks like the issue comes from the `navigationFallback` rules. We are redirecting all requests to `index.html`:
```
 "navigationFallback": {
        "rewrite": "/index.html",
        "exclude": []
    },
```

Updating the rule to the following works:
```
 "navigationFallback": {
        "rewrite": "/index.html",
        "exclude": ["*.{js,map,css,png,ico}"]
    },
```
![image](https://user-images.githubusercontent.com/1699357/114740425-204d2700-9d4a-11eb-9f35-af34bbcde8ab.png)


The CLI intentionally rewrites all routes that don't match the rule. Should we ignore `navigationFallback` if no `exclude` filters were given?

**Comment by manekinekko** [comment]
I've included a fix for this as part of #173. Lmk if this fixes your issue.

https://github.com/Azure/static-web-apps-cli/blob/23699329b607fca74789d6b27d950d5562bc9089/src/proxy/routes-engine/rules/navigationFallback.ts#L27-L30

**Comment by anthonychu** [comment]
I think we need to rearchitect the order in which things are processed (fallback and response overrides act on the response, not request). Haven’t had time to investigate and write up what the rules should be. Will try to do that in a couple of days.

**Comment by runceel** [comment]
@manekinekko 

I'm using the latest SWA CLI v0.3.0. I'm facing different response status codes between Static Web Apps and SWA CLI.
I guess that is related to a PR for this issue.

On SWA(Azure), in unauthorized cases, a response status code is **401**. But SWA CLI returns **403**. I checked the following PowerShell script.
I expect returning **401** on unauthorized cases.

```
## SWA(Azure)
> $SwaResponse = Invoke-WebRequest -Uri "https://brave-bay-031eac500.azurestaticapps.net/" -SkipHttpErrorCheck
> $SwaResponse.StatusCode
401

## SWA CLI
> $SwaCliResponse = Invoke-WebRequest -Uri "http://localhost:4280" -SkipHttpErrorCheck
> $SwaCliResponse.StatusCode
403
```

**Comment by manekinekko** [comment]
@runceel  I think we have fixed this issue in the [main](https://github.com/Azure/static-web-apps-cli/commit/main) branch but we didn't cut a new release yet. We are also improving the routes engine logic (#186). Please stay tuned.

**Comment by runceel** [comment]
@manekinekko Thanks!
I'm looking forward to the release of next version.

**Comment by manekinekko** [comment]
This issue has been fixed by #186
